File: TenXEmpires.Server.Domain/DataContracts/Saves.cs
using System;
using System.Collections.Generic;
using TenXEmpires.Server.Domain.Entities;

namespace TenXEmpires.Server.Domain.DataContracts;

/// <summary>
/// Manual save slot item from <see cref="Save"/>.
/// </summary>
public sealed record SaveManualDto(
    long Id,
    int Slot,
    int TurnNo,
    DateTimeOffset CreatedAt,
    string Name)
{
    public static SaveManualDto From(Save s) => new(s.Id, s.Slot ?? 0, s.TurnNo, s.CreatedAt, s.Name);
}

/// <summary>
/// Autosave item from <see cref="Save"/>.
/// </summary>
public sealed record SaveAutosaveDto(
    long Id,
    int TurnNo,
    DateTimeOffset CreatedAt)
{
    public static SaveAutosaveDto From(Save s) => new(s.Id, s.TurnNo, s.CreatedAt);
}

/// <summary>
/// Response for GET /games/{id}/saves
/// </summary>
public sealed record GameSavesListDto(
    IReadOnlyList<SaveManualDto> Manual,
    IReadOnlyList<SaveAutosaveDto> Autosaves);

/// <summary>
/// Response for POST /games/{id}/saves/manual
/// </summary>
public sealed record SaveCreatedDto(
    long Id,
    int Slot,
    int TurnNo,
    DateTimeOffset CreatedAt,
    string Name)
{
    public static SaveCreatedDto From(Save s) => new(s.Id, s.Slot ?? 0, s.TurnNo, s.CreatedAt, s.Name);
}

/// <summary>
/// Response for POST /saves/{saveId}/load
/// </summary>
public sealed record LoadSaveResponse(
    long GameId,
    GameStateDto State);

File: tenxempires.client/src/types/saves.ts
// Saves types aligned with TenXEmpires.Server.Domain.DataContracts.Saves

import type { GameStateDto } from './game'

// ============================================================================
// Saves DTOs
// ============================================================================

export interface SaveManualDto {
  id: number
  slot: number
  turnNo: number
  createdAt: string
  name: string
}

export interface SaveAutosaveDto {
  id: number
  turnNo: number
  createdAt: string
}

export interface GameSavesListDto {
  manual: SaveManualDto[]
  autosaves: SaveAutosaveDto[]
}

// ============================================================================
// Command DTOs
// ============================================================================

export interface CreateManualSaveCommand {
  slot: number
  name: string
}

// ============================================================================
// Response DTOs
// ============================================================================

export interface SaveCreatedDto {
  id: number
  slot: number
  turnNo: number
  createdAt: string
  name: string
}

export interface LoadSaveResponse {
  gameId: number
  state: GameStateDto
}

// ============================================================================
// View Models (Client-only)
// ============================================================================

export interface OverwriteConfirm {
  slot: 1 | 2 | 3
  oldName?: string
  newName: string
}

